sklearn/utils/__init__.py

```diff
@@ -126,12 +127,29 @@ def _polars_indexing(X, key, key_dtype, axis):
     """Indexing X with polars interchange protocol."""
     # Polars behavior is more consistent with lists
     if isinstance(key, np.ndarray):
+        # Convert each element of the array to a Python scalar
```

Comment:
```suggestion
```
Elements of the ndarray are not touched, only the data container changes from ndarray to list.